Conceptpower+Spring/src/main/java/edu/asu/conceptpower/validation/ConceptTypeAddValidator.java

```diff
@@ -0,0 +1,42 @@
+package edu.asu.conceptpower.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+import edu.asu.conceptpower.db4o.TypeDatabaseClient;
+import edu.asu.conceptpower.web.ConceptTypeAddForm;
+
+@Component
+public class ConceptTypeAddValidator implements Validator {
+
+	@Autowired
+	private TypeDatabaseClient client;
+
+	@Override
+	public boolean supports(Class<?> clazz) {
+		return clazz.isAssignableFrom(ConceptTypeAddForm.class);
+	}
+
+	@Override
+	public void validate(Object obj, Errors errors) {
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "typeName", "required.type_name");
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "typeDescription", "required.type_description");
+
+		if (!errors.hasErrors()) {
+			ConceptTypeAddForm conceptTypeAddForm = (ConceptTypeAddForm) obj;
+			// If the name is same as old name then user has not changed the
+			// concept type name
+			if (!conceptTypeAddForm.getOldTypeName().equalsIgnoreCase(conceptTypeAddForm.getTypeName())) {
```

Comment:
I can do it using jQuery. But once the user starts to change the value and unknowingly he keeps the same old type name, the system will try to update.
Or in JQuery I should also keep a check if user has changed the value by comparing with old value that was loaded in the screen load ?